partitura/performance.py

```diff
@@ -170,19 +171,28 @@ def note_array(self, *args, **kwargs) -> np.ndarray:
         note_array = []
         for n in self.notes:
             note_on_sec = n["note_on"]
+            note_off_sec = n.get("sound_off", n["note_off"])
+            duration_sec = note_off_sec - note_on_sec
+            
+            # tick on, off, duration:
+            # if tick_off is not provided, infer from default mpq and ppq, otherwise use provided)
             note_on_tick = n.get(
                 "note_on_tick",
                 seconds_to_midi_ticks(n["note_on"], mpq=self.mpq, ppq=self.ppq),
             )
-            offset = n.get("sound_off", n["note_off"])
-            duration_sec = offset - note_on_sec
-            duration_tick = (
-                n.get(
+            if n.get("note_off_tick", None) is None:
+                note_off_tick = n.get(
                     seconds_to_midi_ticks(n["sound_off"], mpq=self.mpq, ppq=self.ppq),
-                    seconds_to_midi_ticks(n["note_off"], mpq=self.mpq, ppq=self.ppq),
+                    seconds_to_midi_ticks(n["note_off"], mpq=self.mpq, ppq=self.ppq)
                 )
-                - note_on_tick
-            )
+                duration_tick = n.get(
+                    seconds_to_midi_ticks(n["sound_off"], mpq=self.mpq, ppq=self.ppq),
+                    seconds_to_midi_ticks(n["note_off"], mpq=self.mpq, ppq=self.ppq)
+                ) - note_on_tick
+            else:
+                note_off_tick = n["note_off_tick"]
+                duration_tick = note_off_tick - note_on_tick            
```

Comment:
Do you think it might be worthwhile to add another "layer" to avoid negative durations by e.g., clipping negative durations at this point (e.g., `duration_tick = max(duration_tick, 0)`). On the other hand this might introduce some issues that might be hard to debug.